projects/packages/jetpack-mu-wpcom/src/features/block-patterns/class-wpcom-block-patterns-from-api.php

```diff
@@ -46,53 +46,26 @@ public function register_patterns() {
 		$pattern_categories = array();
 		$block_patterns     = $this->get_patterns( $patterns_cache_key );
 
+		// Register categories from first pattern in each category.
 		foreach ( (array) $block_patterns as $pattern ) {
 			foreach ( (array) $pattern['categories'] as $slug => $category ) {
-				// Register categories from first pattern in each category.
-				if ( ! isset( $pattern_categories[ $slug ] ) ) {
+				// Skip categories that start with an underscore
+				$is_hidden_category = substr( $slug, 0, 1 ) === '_';
+
+				if ( ! isset( $pattern_categories[ $slug ] ) && ! $is_hidden_category ) {
 					$pattern_categories[ $slug ] = array(
 						'label'       => $category['title'],
 						'description' => $category['description'],
 					);
-				}
-			}
-		}
-
-		// Unregister existing categories so that we can insert them in the desired order (alphabetically).
```

Comment:
We don't control the order of categories. That's controlled by the editor. See https://github.com/Automattic/wp-calypso/issues/77130#issuecomment-1935758156.

Reply:
This is new territory for me, so forgive me if I ask silly questions, but why doesn't this affect the order of categories in the Assembler?

Reply:
Because in Assembler we force the order [here](https://github.com/Automattic/wp-calypso/blob/2158a6196cce2b365017718764e84fdc2bc09b1b/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-categories-order.ts#L9)